javagen/src/main/java/com/azure/autorest/mapper/ClientMapper.java

```diff
@@ -221,54 +230,64 @@ public Client map(CodeModel codeModel) {
         // async/sync service client (wrapper for the ServiceClient)
         List<AsyncSyncClient> syncClients = new ArrayList<>();
         List<AsyncSyncClient> asyncClients = new ArrayList<>();
-        if (settings.isGenerateSyncAsyncClients()) {
-            ClientModelUtil.getAsyncSyncClients(codeModel, serviceClient, asyncClients, syncClients);
-        }
-        builder.syncClients(syncClients);
-        builder.asyncClients(asyncClients);
-
-        // service client builder
-        if (!serviceClient.builderDisabled()) {
-            List<ClientBuilder> clientBuilders = new ArrayList<>();
-            String builderSuffix = ClientModelUtil.getBuilderSuffix();
-            String builderName = serviceClient.getInterfaceName() + builderSuffix;
-            String builderPackage = ClientModelUtil.getServiceClientBuilderPackageName(serviceClient);
-            if (settings.isGenerateSyncAsyncClients() && settings.isGenerateBuilderPerClient()) {
-                // service client builder per service client
-                for (int i = 0; i < asyncClients.size(); ++i) {
-                    AsyncSyncClient asyncClient = asyncClients.get(i);
-                    AsyncSyncClient syncClient = (i >= syncClients.size()) ? null : syncClients.get(i);
-                    String clientName = ((syncClient != null)
-                            ? syncClient.getClassName()
-                            : asyncClient.getClassName().replace("AsyncClient", "Client"));
-                    String clientBuilderName = clientName + builderSuffix;
-                    ClientBuilder clientBuilder = new ClientBuilder(
-                            builderPackage, clientBuilderName, serviceClient,
-                            (syncClient == null) ? Collections.emptyList() : Collections.singletonList(syncClient),
-                            Collections.singletonList(asyncClient));
-
+        List<ClientBuilder> clientBuilders = new ArrayList<>();
+        for (Map.Entry<ServiceClient, ? extends com.azure.autorest.extension.base.model.codemodel.Client> entry : serviceClientsMap.entrySet()) {
```

Comment:
mostly refactor, move it under a loop to support List of ServiceClient

logic not changed, create a ClientBuilder, list of sync/async wrapper AsyncSyncClient for one ServiceClient.